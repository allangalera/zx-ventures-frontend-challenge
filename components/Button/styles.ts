export const Button = (theme: any) => ({
  color: 'white',
  background: 'black',
  border: 'none',
  borderRadius: '0.5rem',
  padding: '1rem 2rem',
});

export const ButtonSpan = (theme: any) => ({});
