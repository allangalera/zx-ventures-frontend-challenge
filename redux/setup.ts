import { createStore, combineReducers, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

import cartReducer from '@redux/cart';

const rootReducer = combineReducers({
  cart: cartReducer,
});

const store = createStore(
  rootReducer,
  composeWithDevTools(),
  // applyMiddleware(...middleware),
  // other store enhancers if any
);

export default store;
