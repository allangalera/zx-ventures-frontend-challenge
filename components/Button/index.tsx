import React from 'react';
import * as styles from './styles';

type Props = {
  text: string;
  handleClick: Function;
};

const Button: React.FunctionComponent<Props> = ({ text, handleClick }): JSX.Element => {
  const onClick = () => {
    console.log('aqui');
    handleClick();
  };
  return (
    <button css={styles.Button} onClick={onClick}>
      <span css={styles.ButtonSpan}>{text}</span>
    </button>
  );
};

export default Button;
