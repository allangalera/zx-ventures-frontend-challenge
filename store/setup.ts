import { createStore, combineReducers, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from '@store/combineReducers';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export default () => {
  let store = createStore(
    persistedReducer,
    composeWithDevTools(),
    // applyMiddleware(...middleware),
    // other store enhancers if any
  );
  let persistor = persistStore(store);
  return { store, persistor };
};
