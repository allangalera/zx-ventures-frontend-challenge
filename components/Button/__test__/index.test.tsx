import React from 'react';
import { render, fireEvent } from '@test/setup';
import Button from '@components/Button';

describe('Button component', () => {
  const defaultProps = {
    handleClick: jest.fn(),
    text: 'Button Label',
  };

  it('matches snapshot', () => {
    const { asFragment } = render(<Button {...defaultProps} />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it('click triggers', () => {
    const { getByText } = render(<Button {...defaultProps} />, {});
    fireEvent.click(getByText(defaultProps.text));
    expect(defaultProps.handleClick).toBeCalled();
  });
});
