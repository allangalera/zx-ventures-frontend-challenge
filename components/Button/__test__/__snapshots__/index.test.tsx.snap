// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="css-1gk7bz7-Button"
  >
    <span
      class="css-1dis8g1-Button"
    >
      Button Label
    </span>
  </button>
</DocumentFragment>
`;

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="css-1gk7bz7-Button"
  >
    <span
      class="css-1dis8g1-Button"
    >
      Button Label
    </span>
  </button>
</DocumentFragment>
`;
